src/home/homework.c

```diff
@@ -48,6 +48,10 @@ int user_record_authenticate(
                 PasswordCache *cache,
                 bool strict_verify) {
 
+        /* We use a single scratch buffer for all password testing and securely destroy it at the end */
+        _cleanup_(erase_and_freep) void *cd_data = NULL;
+        int cd_size = 0;
```

Comment:
i'd just hide this internally. a malloc() doesn#t count. and a memzero() doesn't matter either... This is not inner loop stuff, and thus reusing these buffers is premature optimization. Let's encapsulate this nicely, and keep the buffers in the innermost code block only...